.github/workflows/run-unit-on-fork.yml

```diff
@@ -0,0 +1,30 @@
+# Description: This workflow runs unit test on PRs created from forks.
+#
+# Triggered by: Every commit on the PR.
+
+name: Test / Fork / Unit
+
+concurrency:
+  group: unit-fork-${{ github.event.pull_request.id }}
+  cancel-in-progress: true
+
+on:
+  pull_request:
```

Comment:
Note: this workflow will not have write permissions or access to any secrets ([docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflows-in-forked-repositories-1)). This is okay for unit test since it doesn't need any secrets anyway.